Clang-based SCA tools fail on Atmel HAL Boards with C++ code

**Describe the bug**

When running SCA with clang-based tools (such as `clangsa`) on C++ code that targets Atmel HAL boards, the analysis fails.  
Clang does not compile certain parts of the Atmel HAL code.

**To Reproduce**

1. Install [CodeChecker](https://docs.zephyrproject.org/latest/develop/sca/codechecker.html)
2. Install clang to have `clangsa` available
3. Build `cpp_synchronization` sample with SCA for an Atmel HAL board
```
west build -p -b same54_xpro samples/cpp/cpp_synchronization/ -- -DZEPHYR_SCA_VARIANT=codechecker
```
Clang trows `anonymous bit-field cannot have qualifiers` error (see log)

_Note:_  
To reduce the output from CodeChecker I added `-DCODECHECKER_ANALYZE_OPTS="--analyzers;clangsa;--ignore;../sca_ignore.txt"` with the following ignore file:
```
+*/samples/*
-*/*
```

**Expected behavior**
`clangsa` analyzes the code.

**Impact**
Can't use SCA to analyze C++ projects with clang-based tools for boards that use the Atmel HAL.

**Logs and console output**
Output of west build with SCA (see above):
```
...
[INFO 2025-01-20 14:37] - Starting static analysis ...
[ERROR 2025-01-20 14:37] - Analyzing main.cpp with clangsa  failed!
[ERROR 2025-01-20 14:37] - 

[ERROR 2025-01-20 14:37] - 
In file included from /home/user/zephyrproject/zephyr/samples/cpp/cpp_synchronization/src/main.cpp:22:
In file included from /home/user/zephyrproject/zephyr/include/zephyr/kernel.h:17:
In file included from /home/user/zephyrproject/zephyr/include/zephyr/kernel_includes.h:36:
In file included from /home/user/zephyrproject/zephyr/include/zephyr/arch/cpu.h:19:
In file included from /home/user/zephyrproject/zephyr/include/zephyr/arch/arm/arch.h:30:
In file included from /home/user/zephyrproject/zephyr/include/zephyr/arch/arm/asm_inline.h:18:
In file included from /home/user/zephyrproject/zephyr/include/zephyr/arch/arm/asm_inline_gcc.h:24:
In file included from /home/user/zephyrproject/zephyr/modules/cmsis/cmsis_core.h:10:
In file included from /home/user/zephyrproject/zephyr/modules/cmsis/cmsis_core_m.h:24:
In file included from /home/user/zephyrproject/zephyr/soc/atmel/sam0/same54/soc.h:24:
In file included from /home/user/zephyrproject/modules/hal/atmel/asf/sam0/include/same54/same54p20a.h:582:
/home/user/zephyrproject/modules/hal/atmel/asf/sam0/include/same54/component/ac.h:232:18: error: anonymous bit-field cannot have qualifiers
  232 |     __I uint8_t  :2;               /*!< bit:  2.. 3  Reserved                           */
      |                  ^
...
```

**Environment:**

 - OS: Linux
 - Toolchain: Zephyr SDK
 - Zephyr v4.0.0
 - Clang 18.1.8
 - CodeChecker 6.24.4

**Additional context**
If I try the `cpp/hello_world` sample, it works without errors.   
Also, non-C++ samples can be analyzed without problems.


It seems that the problem occurs because `main.cpp` in the `cpp_synchronization` -Sample included `zephyr/kernel.h`. This caused the HAL headers to be compiled with **clang in c++**.  
Clang seems to be more strict than GCC regarding the anonymous bit-fields used in the Atmel HAL headers, and fails to compile.

_Note:_
In addition to that, the Atmel HAL also has bit-fields which are named `EMPTY` ([in dac.h](https://github.com/search?q=repo%3Azephyrproject-rtos%2Fhal_atmel++%22+EMPTY%3A%22&type=code)) which collides with the [zephyr utility macro](https://docs.zephyrproject.org/apidoc/latest/group__sys-util.html#ga2b7cf2a3641be7b89138615764d60ba3). This results in more unnamed bit-field errors.

--- a/west.yml
+++ b/west.yml
@@ -157,7 +157,7 @@ manifest:
       groups:
         - hal
     - name: hal_atmel
-      revision: dde7a36d00bd5dafd1ffcd8d243dfbceeb20e856
+      revision: d697369628ba67d4613b90aa61a2822c3aaa876e
       path: modules/hal/atmel
       groups:
         - hal

